homeassistant/components/zha/switch.py

```diff
@@ -236,14 +254,29 @@ def inverted(self) -> bool:
     @property
     def is_on(self) -> bool:
         """Return if the switch is on based on the statemachine."""
-        val = bool(self._cluster_handler.cluster.get(self._attribute_name))
+        if self._on_value != 1:
+            val = self._cluster_handler.cluster.get(self._attribute_name)
+            val = val == self._on_value
+        else:
+            val = bool(self._cluster_handler.cluster.get(self._attribute_name))
         return (not val) if self.inverted else val
 
     async def async_turn_on_off(self, state: bool) -> None:
         """Turn the entity on or off."""
-        await self._cluster_handler.write_attributes_safe(
-            {self._attribute_name: not state if self.inverted else state}
-        )
+        if self.inverted:
+            state = not state
+        if state and self._on_value != 1:
+            await self._cluster_handler.write_attributes_safe(
+                {self._attribute_name: self._on_value}
+            )
+        elif not state and self._off_value != 0:
+            await self._cluster_handler.write_attributes_safe(
+                {self._attribute_name: self._off_value}
+            )
+        else:
+            await self._cluster_handler.write_attributes_safe(
+                {self._attribute_name: state}
+            )
```

Comment:
The extra check (and not always writing `_on_value` / `_off_value`) is done to make sure the type is a `bool`, right?
Is there really any difference though (or is this done bc of some other reason)? zigpy should always use the correct "Zigbee type" at least.

Reply:
no these are to use custom on and off values... meaning we have to write something non 0 or non 1 to the device to make it on or off. there is a test for this.

Reply:
Yes, but why do we need the third condition for that? `_on_value` and `_off_value` are 1 and 0 by default. We can just always write `_on_value` (or `_off_value`).
I would think this logic could be simplified.

(Might be missing something though, I'll have another look at this later again..)

Reply:
Adjusted